backend/src/main/java/com/google/minicrm/data/Lead.java

```diff
@@ -153,9 +167,39 @@ public Entity asEntity() {
     leadEntity.setProperty("creativeId", creativeId);
     leadEntity.setProperty("notes", notes);
     leadEntity.setProperty("status", status.ordinal());
+    List<String> locationInfos = new ArrayList<String>();
     for (String key : columnData.keySet()) {
       leadEntity.setProperty(key, columnData.get(key));
+      // If the column data is related with address add it into locationInfo arraylist
+      if (key.equals("POSTAL_CODE") || key.equals("STREET_ADDRESS") || key.equals("CITY") || key.equals("REGION") || key.equals("COUNTRY") || key.equals("COUNTRY") ){
+        locationInfos.add(columnData.get(key));
+      }
+    }
+
+    // Stringfy the list
+    String addressToBe = "";
```

Comment:
since you are doing repeated concatenations in a loop, use StringBuilder instead with append.